webpack/assets/javascripts/react_app/components/HostsIndex/index.js

```diff
@@ -233,12 +245,38 @@ const HostsIndex = () => {
       {__('Build management')}
     </MenuItem>,
     <MenuItem
-      itemId="reassign-hg-dropdown-item"
-      key="reassign-hg-dropdown-item"
-      onClick={setHgModalOpen}
+      itemId="host-affiliation-dropdown-item"
+      key="host-affiliation-dropdown-item"
       isDisabled={selectedCount === 0}
+      flyoutMenu={
+        <Menu
+          ouiaId="host-affiliation-dropdown-menu"
+          onSelect={() => setMenuOpen(false)}
+        >
+          <MenuContent>
+            <MenuList>
+              <MenuItem
+                itemId="reassign-hg-dropdown-item"
+                key="reassign-hg-dropdown-item"
+                onClick={setHgModalOpen}
+                isDisabled={selectedCount === 0}
+              >
+                {__('Host group')}
+              </MenuItem>
+              <MenuItem
+                itemId="assign-taxonomy-dropdown-item"
+                key="assign-taxonomy-dropdown-item"
+                onClick={setTaxonomyModalOpen}
+                isDisabled={selectedCount === 0}
+              >
+                {__('Assign organization/ location')}
```

Comment:
The space after the `/` is really jarring to me. I also question `assign`. We don't have that for `Host group` so why do we have it here? Would this be better?
```suggestion
                {__('Change organization and/or location')}
```

Reply:
I agree with you. And in my opinion it is not a good idea to combine organization and location into one page.  But I might be wrong.
 
Let me paste the mock page here that I'm trying to follow.
![m1](https://github.com/user-attachments/assets/6c60b4de-10a0-4f7a-a9f4-1808b097216d)

Reply:
I debated that as well, but there is one potential benefit: if the location belongs to another org you can do it in a single operation. It may otherwise not be possible.

Another option is to go another route and actually combine all bulk changes into 1: also include the host group into the dialog. Then the menu item is simple (a single button). Host groups may also belong to another org so it may actually be common to perform that in a single operation.

Reply:
It is a multiple operation to change the hosts if both organization and location are selected. Hosts are modified in separate calls.
It's not just two attributes. There is lots of [checking](https://github.com/theforeman/foreman/blob/a3962dfdc385c9a3786062ae404b325cad8e89e4/app/services/tax_host.rb#L92) there for a list of things that might mismatch. 
Without enough knowledge in that area, I think it's better just keep it as it is.